core/modules/user/user.module

```diff
@@ -815,6 +829,13 @@ function user_account_form(&$form, &$form_state) {
       '#type' => 'password',
       '#password_toggle' => TRUE,
       '#password_strength' => TRUE,
+      '#attributes' => array(
+        // Add attributes to field to prevent spell-jacking and unwanted
+        // automatic changes caused by the browser.
+        'autocapitalize' => 'none',
+        'autocorrect' => 'off',
+        'spellcheck' => 'false',
+      ),
```

Comment:
This block simply adds the desired attributes to both name and password for this specific high-level form.